components/application-broker/internal/broker/testing/client_assertion.go

```diff
@@ -50,11 +54,21 @@ func (a *ActionsAsserter) AssertCreates(t *testing.T, expect []runtime.Object) {
 			t.Errorf("Missing create: %#v", want)
 			continue
 		}
-
 		got := a.Creates[i].GetObject()
 
-		if diff := cmp.Diff(want, got); diff != "" {
-			t.Errorf("Unexpected create (-want, +got): %s", diff)
+		wantType := reflect.TypeOf(want).String()
```

Comment:
why do a string compare here?
wantType.(type) == v1beta1.PeerAuthentication ... should also work and make it more readable